katdal/test/test_chunkstore_rados.py

```diff
@@ -26,29 +26,22 @@
 
 
 class TestRadosChunkStore(ChunkStoreTestBase):
-    """Tests connecting to an actual RADOS service, implying a slower setup."""
+    """Test Ceph functionality by connecting to an actual RADOS service."""
 
-    def setup(self):
+    @classmethod
+    def setup_class(cls):
         # Look for default Ceph installation but expect a special test pool
         config = '/etc/ceph/ceph.conf'
         pool = 'test_katdal'
         try:
-            self.store = RadosChunkStore.from_config(config, pool)
+            cls.store = RadosChunkStore.from_config(config, pool)
         except (ImportError, StoreUnavailable):
             raise SkipTest('Rados not installed or cluster misconfigured / down')
 
     def array_name(self, path):
         namespace = 'katdal_test_chunkstore_rados'
         return self.store.join(namespace, path)
 
-
-class TestDudRadosChunkStore(object):
-    """Tests that don't need a RADOS connection, only a 'dud' store."""
-
-    def setup(self):
-        if not rados:
-            raise SkipTest('Rados not installed')
-
     def test_store_unavailable(self):
         # Pretend that rados is not installed
         katdal.chunkstore_rados.rados = None
```

Comment:
With this no longer being in a separate TestDudRadosChunkStore class, won't it cause things to break if the test runner decides to run this test before the others?

Reply:
It restores the module three lines down, so should be idempotent.

Reply:
Thanks, that hadn't shown up the diff.

Reply:
It should probably be done in a try/finally block though, so that if the test fails it still restores the previous state.

Reply:
Done.